lib/template-parts.php

```diff
@@ -213,6 +213,20 @@ function filter_rest_wp_template_part_query( $args, $request ) {
 			'value' => $request['theme'],
 		);
 
+		// Ensure auto-drafts of all theme supplied template parts are created.
+		if( $request['theme'] === get_current_theme() ) {
+			$template_part_files = glob( get_stylesheet_directory() . '/block-template-parts/*.html' );
+			$template_part_files = is_array( $template_part_files ) ? $template_part_files : array();
+			if ( is_child_theme() ) {
+				$child_template_part_files = glob( get_template_directory() . '/block-template-parts/*.html' );
+				$child_template_part_files = is_array( $child_template_part_files ) ? $child_template_part_files : array();
+				$template_part_files       = array_merge( $template_part_files, $child_template_part_files );
+			}
+			foreach ( $template_part_files as $template_part ) {
+				create_auto_draft_for_template_part_block( $template_part );
```

Comment:
This function takes a block, not a file.

Reply:
Ahh yeah.  And it looks like the template part files only contain the inner content.  That is, the template files contain the template part definitions that have the slug/theme name, but the template part files do not contain that information inside themselves.

So im getting the content from the template part file, and wrapping it with a template-part block string and parsing it into a block.  Im filling in that missing info by assuming the slug attribute will always correspond to the file name and the theme attr will always correspond to the currently applied theme in this case.  I think thats fair?

Does this approach make more sense? (pushed these changes noted above)

Reply:
After stepping away and thinking about it more, would it make more sense to do the query and create post here?  As opposed to building a string to parse it into a block to give to a function that is going to go back to the stylesheet directory again anyways?

Reply:
It'd be nice to extract the logic for creating the post so that it can be referenced in both places.